CMakeLists.txt

```diff
@@ -230,34 +231,44 @@ if(ENABLE_QT)
     set(ENABLE_QT_COM_INTEROP OFF)
     set(ENABLE_QT_DBUS_INTEROP OFF)
 
-    set(QT5_REQUIRED_MODULES Core Gui Widgets Network LinguistTools)
-    set(QT5_OPTIONAL_MODULES DBus AxContainer AxServer)
+    set(QT_REQUIRED_MODULES Core Gui Widgets Network LinguistTools)
+    set(QT_OPTIONAL_MODULES DBus AxContainer AxServer)
+    set(MISSING_QT_MODULE)
 
-    if(WIN32)
-        list(APPEND QT5_REQUIRED_MODULES WinExtras)
+    set(Qt_NAMES Qt6 Qt5)
+    if(NOT USE_QT_VERSION STREQUAL "AUTO")
+        set(Qt_NAMES Qt${USE_QT_VERSION})
     endif()
 
-    foreach(M ${QT5_REQUIRED_MODULES})
-        find_package(Qt5${M} ${QT5_MINIMUM} QUIET)
-        if(Qt5${M}_FOUND)
-            if(NOT M STREQUAL "LinguistTools")
-                list(APPEND QT_TARGETS Qt5::${M})
-            endif()
-        else()
-            set(QT_TARGETS)
-            break()
+    find_package(Qt NAMES ${Qt_NAMES} ${QT_MINIMUM} QUIET)
+    if(Qt_FOUND)
+        if(WIN32 AND Qt_VERSION_MAJOR EQUAL 5)
+            list(APPEND QT_REQUIRED_MODULES WinExtras)
         endif()
-    endforeach()
+
+        foreach(M ${QT_REQUIRED_MODULES})
+            find_package(Qt${Qt_VERSION_MAJOR}${M} ${QT_MINIMUM} QUIET)
+            if(Qt${Qt_VERSION_MAJOR}${M}_FOUND)
+                if(NOT M STREQUAL "LinguistTools")
+                    list(APPEND QT_TARGETS Qt${Qt_VERSION_MAJOR}${QT_VERSION_MAJOR}::${M})
```

Comment:
I don't understand`Qt${Qt_VERSION_MAJOR}${QT_VERSION_MAJOR}`.  What is going on here?

Reply:
Oops. Variable names are case-sensitive, so first one expands to "6" while second one — to "", so it works :)